flask/src/main/groovy/net/corda/gradle/flask/FlaskPlugin.groovy

```diff
@@ -0,0 +1,69 @@
+package net.corda.gradle.flask
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.plugins.JavaApplication
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.SourceSet
+import org.gradle.jvm.tasks.Jar
+
+class FlaskPlugin implements Plugin<Project> {
+
+    private static final String GROUP = "Flask"
+
+    @Override
+    void apply(Project project) {
+        project.getPluginManager().apply(JavaPlugin.class)
+        Provider<Directory> flaskDir = project.layout.buildDirectory.dir("classes/flask-launcher")
+        Provider<Copy> extractLauncherTarProvider = project.tasks.register("extractLauncherTar", Copy) {
+            setGroup(GROUP)
+            setDescription("Extract the Flask Launcher classes to be used to build a custom launcher")
+            into(flaskDir)
+            from(project.tarTree(LauncherResource.instance))
+        }
+
+        JavaPluginConvention javaPluginConvention = project.convention.getPlugin(JavaPluginConvention.class)
+        SourceSet flaskSourceSet = javaPluginConvention.sourceSets.create("flask")
+        flaskSourceSet.compiledBy(extractLauncherTarProvider)
+        project.configurations.getByName(flaskSourceSet.compileOnlyConfigurationName).withDependencies { dependencies ->
+            def launcherDependency = project.dependencies.create(extractLauncherTarProvider.get().outputs.files)
+            dependencies.add(launcherDependency)
+        }
+
+        Provider<FlaskJarTask> flaskJarTask = project.tasks.register("flaskJar", FlaskJarTask.class) {
+            setGroup(GROUP)
+            setDescription("Package the current project code in an executable jar file")
```

Comment:
These should both go in the `FlaskJarTask` class itself so that _every_ instance has this group and description.

Reply:
I think it is better to have the description here (I can probably move the group to `FlaskJarTask`), this way the default instance of `FlaskJarTask`, `flaskJar`, will contain that description but any custom instance of `FlaskJarTask` the user may decide to add will have a different description, since the current descrption 
```
"Package the current project code in an executable jar file"
```
actually refers to the `flaskJar` task while a user might want to create an additional instance of that task that creates an artifact with a different JavaAgent or something like that

Reply:
The `FlaskJarTask` needs both a group and a general description, but I suppose that doesn't prevent you from providing this particular task instance with a more specific description.

Reply:
👍🏿

Reply:
damn.. I can't put the group as a memeber of `net.corda.gradle.flask.FlaskPlugin` and then use it in `net.corda.gradle.flask.FlaskJarTask` because `FlaskPlugin` is written in Groovy while `FlaskJarTask` is in Java and `compileJava` is invoked before `compileGroovy` 🤦🏿 

I will add it as a constant in `net.corda.flask.common.Flask`